tests/basic/podmetric_test.go

```diff
@@ -79,10 +120,129 @@ var _ = Describe("{PodMetricFunctional}", func() {
+
 	logglyToken, ok := os.LookupEnv("LOGGLY_API_TOKEN")
 	dash.VerifyFatal(ok, true, "failed to fetch loggly api token")
 	headers := make(map[string]string)
 	headers["Authorization"] = fmt.Sprintf("Bearer %v", logglyToken)
 	return rest.Get(fmt.Sprintf("%v?%v", logglyIterateUrl, query), nil, headers)
 }
+
+func getClusterID() (string, error) {
+	workerNode := node.GetWorkerNodes()[0]
+	clusterID, err := Inst().N.RunCommand(workerNode, fmt.Sprintf("cat %s", "/etc/pwx/cluster_uuid"), node.ConnectionOpts{
+		IgnoreError:     false,
+		TimeBeforeRetry: defaultRetryInterval,
+		Timeout:         defaultTimeout,
+		Sudo:            true,
+	})
+	if err != nil {
+		return "", fmt.Errorf("failed to get pxctl status, Err: %v", err)
+	}
+
+	return clusterID, nil
+}
+
+func getMeteringData(clusterUUID string, meteringInterval time.Duration) ([]*CallhomeData, error) {
+	log.InfoD("fetching logs from loggly")
+
+	lookbackInterval := meteringInterval + 1*time.Minute
+	data, code, err := getLogglyData(clusterUUID, fmt.Sprintf("-%vm", math.Round(lookbackInterval.Minutes())))
+	if err != nil {
+		return nil, err
+	}
+	dash.VerifyFatal(code, 200, fmt.Sprintf("loggly return code %v not equal to 200", code))
+	dash.VerifyFatal(len(data) == 0, false, "loggy return empty response")
+
+	log.InfoD("parsing logs from loggly")
+	var logglyPayload LogglyPayload
+	err = json.Unmarshal(data, &logglyPayload)
+	if err != nil {
+		return nil, err
+	}
+
+	var callhomeEvents []*CallhomeData
+	for _, e := range logglyPayload.Events {
+		chd := CallhomeData{}
+		err = json.Unmarshal([]byte(e.Raw), &chd)
+		if err != nil {
+			return nil, err
+		}
+		callhomeEvents = append(callhomeEvents, &chd)
+	}
+
+	var meteringData []*CallhomeData
+	for _, d := range callhomeEvents {
+		if d.UsageType == "meteringData" {
+			meteringData = append(meteringData, d)
+		}
+	}
+
+	return meteringData, nil
+}
+
+// getExpectedPodHourInMinutes returns the estimate pod hour given that the metering interval is
+// 1 min. it checks a list of volumes and the number of pods using it to estimate the pod hour.
+func getExpectedPodHourInMinutes(contexts []*scheduler.Context, meteringInterval time.Duration) int {
+	var totalPods []v1.Pod
+	for _, ctx := range contexts {
+		log.InfoD("getting pod hour for context %v", ctx.App.Key)
+		vols, err := Inst().S.GetVolumes(ctx)
+		log.FailOnError(err, "Failed to get volumes to check pod hour")
+
+		for _, vol := range vols {
+			pods, err := Inst().S.GetPodsForPVC(vol.Name, vol.Namespace)
+			log.FailOnError(err, "Failed to get pods from PVC")
+			totalPods = append(totalPods, pods...)
+		}
+	}
+
+	// Count one minute per pod using a PX volume
+	return len(totalPods) * int(meteringInterval.Minutes())
+}
+
+func getLatestPodHours(meteringData []*CallhomeData) float64 {
+	return meteringData[0].PodHour
+}
+
+func verifyPodHourWithError(actualPodHours, expectedPodHours, reasonableErrorPercent float64) {
```

Comment:
Same here, can we return errors and do all the error checking in the test? Otherwise it looks very weird looking at test code, where half the functions you are using return error and you handle it in test code and the other half do not return anything and are being handled inside the actual function. We should pick one approach and stick to it. In my opinion all functions should return errors and then handled by the test itself, what do you think?

Reply:
That sounds good to me, I wasn't aware of any existing practices here. Fine with that.